#
# the test is is doomed to fail on Oracle Solaris and other OSes,
# which don't provide FSTYPENAME member in statfs structure.
# patch has been sent to upstream:
#	https://github.com/proftpd/proftpd/pull/705
#

diff --git a/tests/api/fsio.c b/tests/api/fsio.c
index 508ca4657..72fad85c5 100644
--- a/tests/api/fsio.c
+++ b/tests/api/fsio.c
@@ -4637,7 +4637,9 @@ Suite *tests_get_fsio_suite(void) {
   tcase_add_test(testcase, fs_fgetsize_test);
   tcase_add_test(testcase, fs_fadvise_test);
   tcase_add_test(testcase, fs_have_access_test);
+#if defined(HAVE_STATFS_F_TYPE) || defined(HAVE_STATFS_F_FSTYPENAME)
   tcase_add_test(testcase, fs_is_nfs_test);
+#endif
   tcase_add_test(testcase, fs_valid_path_test);
   tcase_add_test(testcase, fsio_smkdir_test);
   tcase_add_test(testcase, fsio_getpipebuf_test);
